src/common/hashes/shamap.ts

```diff
@@ -0,0 +1,288 @@
+import hashPrefix from './hash-prefix'
+import hash from './sha512hash'
+const HEX_ZERO = '0000000000000000000000000000000000000000000000000000000000000000'
+
+export enum NodeTypes {
+  INNER = 1,
+  TRANSACTION_NM = 2,
```

Comment:
I'd consider dropping abbreviations. It's not immediately obvious what `NM` or `MD` are.

Reply:
@sublimator  Do you know what `MD` or `NM` stand for here?

Reply:
NM = No metadata (used in consensus for proposed transaction sets iirc), and MD = transactions with metadata (as per a closed|validated ledger)